modules/debugging/overview.md

```diff
@@ -1,23 +1,229 @@
+case for everything.
+
+Note 2: If it blows up in the empty case
+This is often the most overlooked case, but make sure something happens if it's empty
+it will be accounted for. 
+
+Note 3: Incorrectly assuming that there will be no exceptional cases
+There can be anything inputted. Either limit what you are getting inputted OR account 
+for it in your code. Anything that can be accepted as a data point should be accounted for.
+
+
+### Extraneous Assumption
+Description:
+This is when you assume certain things about the input data that may not be true. It 
+usually means you have to change or account for certain things in your data and maybe even
+*clean* your data to make sure it can and will run properly. 
+
+Ways to tell this is the problem:
+1. Resulting data or output is not in the right form.  
+
+Note:
+Never assume that something not specified in the parameters is given to you. Often
+times you will need to make sure everything passed in to you, or any data you receive is
+in the correct form that you expect it to be before working with it.
+
+1. An ordered list
+2. No duplicates of data
+3. Correct or accurate data
+
+### Incorrect order of Input 
+Description:
+The input of the data is not in the order you expected. This could be in many forms
+such as items in a list, order of data in a string, etc.
+
+Ways to tell this is the problem:
+1. Type errors usually inform you of this.
+2. Data has not been changed from original form.
+
+Note:
+Make sure you reorder your input to make sure if fits what you are excpecting before
+you proceed with working with it. 
+
+<!-- ### Incorrect order of Results ISNT THIS SORT OF AN OBVIOUS ERROR THAT DOES'T REALLY REQUIRE CHANGE-->
+
+### Off by one error
+Description:
+These are usually very easy to spot but you are consistantly off by one in your 
+measure or test in something.
+
+Note:
+Generally there is an edge case that is missed or an index that is not being accounted for.
+If there is a patter of always being off by one in your result that means you are probably missing the first
+or last case.
+
+### Incorrect level of Nesting 
+Description:
+This usually means in a nxn object exactly an (n-1)x(n-1) amount of data is being changed or 
+used. 
+
+Ways to tell:
+1. Only a part of your data (usually the first row or column) has been changed.
+
+
+<!-- ### Misnamed property DONT THINK THIS IS NECESSARY PERSONALLY-->
+
+### Incorrect Mathematical Calculation
+Description:
+This is a common mistake that can step from many things, but you should always make sure it 
+works for all possible inputs.
+
+Ways to tell:
+1. Your math doesn't output the right number.
+
+Note: This can be from:
+1. Confusing median for mean. 
+2. Divide by zero.
+3. Error on negative numbers, or 0.
+
+<!-- ### typo or reversal in translating a formula NOT SURE I FOLLOW THIS ONE-->
+
+<!-- ### incorrectly consuming an api response object WON'T YOUR FETCH/GET REQUEST TELL YOU WHAT YOU GET AND 
```

Comment:
Assuming that the response comes in as a JSON object, first there is no type information in JSON, second, by default nobody knows what is "correct" because it's just some arbitrary data-blob.  Correctness is a concern of the problem domain, which a language or primitive types don't really help with.